models/classes/dataBinding/class.GenerisInstanceDataBinder.php

```diff
@@ -94,23 +88,22 @@ protected function getTargetInstance()
      * @access public
      * @author Jerome Bogaerts, <jerome@taotesting.com>
      * @param  array data An array of values where keys are Property URIs and values are either scalar or vector values.
+     * @throws tao_models_classes_dataBinding_GenerisInstanceDataBindingException
      * @return mixed
      */
     public function bind($data)
     {
-        $returnValue = null;
-
         try {
             $instance = $this->getTargetInstance();
-            $eventManager = \oat\oatbox\service\ServiceManager::getServiceManager()->get(\oat\oatbox\event\EventManager::CONFIG_ID);
+            $eventManager = $this->getEventManager();
             foreach ($data as $propertyUri => $propertyValue) {
                 if ($propertyUri == OntologyRdf::RDF_TYPE) {
                     foreach ($instance->getTypes() as $type) {
                         $instance->removeType($type);
                     }
-                    if (!is_array($propertyValue)) {
-                        $types = [$propertyValue] ;
-                    }
+
+                    $types = !is_array($propertyValue) ? [$propertyValue] : $propertyValue;
```

Comment:
I would avoid negation always as possible, since it is less readable.

```suggestion
                    $types = is_array($propertyValue) ? $propertyValue : [$propertyValue];
```

Reply:
Changed :point_right: 83663b774db2ba839cbd60fef93d53856fde82b7 :+1: